### PR TITLE
Imlement most missing rxjs libdefs

### PR DESCRIPTION
**Pull request by Kriegslustig**
This PR extends the rxjs libdefs, so that they deflect the official documentation. I did find a few cases where they defined a type signature loosesly because of restrictions in TypeScript. In those cases I tried to be as restrictive as possible.

### CODE REVIEW

File: definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ interface rxjs$IObserver<-T> {
   complete(): mixed;
 }
 
-type rxjs$PartialObserver<-T> =
+declare type rxjs$PartialObserver<-T> =
```
Comment (nmote) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js]: Why the declare here?

File: definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js [hunk 2]
```diff
@@ -166,18 +231,32 @@ declare class rxjs$Observable<+T> {
   mapTo<U>(value: U): rxjs$Observable<U>;
 
   merge(other: rxjs$Observable<T>): rxjs$Observable<T>;
+  merge(others: Iterable<rxjs$Observable<T>>): rxjs$Observable<T>;
+  merge(...others: Array<rxjs$Observable<T>>): rxjs$Observable<T>;
+
+  static merge(other: rxjs$Observable<T>): rxjs$Observable<T>;
```
Comment (nmote) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js]: I'm pretty sure the existing typedefs for `static merge` are more precise. Could you maintain that precision or add tests to prove that something like `Observable.merge(numbers, strings)` has type `Observable<number | string>`

File: definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js [hunk 3]
```diff
@@ -166,18 +231,32 @@ declare class rxjs$Observable<+T> {
...
+  static merge(others: Iterable<rxjs$Observable<T>>): rxjs$Observable<T>;
+  static merge(...others: Array<rxjs$Observable<T>>): rxjs$Observable<T>;
 
   mergeAll<U>(): rxjs$Observable<U>;
 
   mergeMap<U>(
-    project: (value: T, index?: number) => rxjs$Observable<U> | Promise<U> | Iterable<U>,
+    p: (value: T, index?: number) => rxjs$Observable<U> | Promise<U> | Iterable<U>,
```
Comment (nmote) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js]: I'd prefer more descriptive parameter names throughout.

File: definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js [hunk 4]
```diff
@@ -213,30 +292,36 @@ declare class rxjs$Observable<+T> {
...
+    | (
+      next?: (value: T) => mixed,
+      error: (error: any) => mixed,
+      complete?: () => mixed,
+    ) => rxjs$Observable<T>
+    | (
+      next?: (value: T) => mixed,
+      error?: (error: any) => mixed,
+      complete: () => mixed,
+    ) => rxjs$Observable<T>;
```
Comment (nmote) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js]: Why do you need all these as well as the `PartialObserver` variant below?

File: definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js [hunk 5]
```diff
@@ -247,35 +332,35 @@ declare class rxjs$Observable<+T> {
...
-    onCompleted: ?() => mixed,
+    next: ?(value: T) => mixed,
+    error: ?(error: any) => mixed,
+    complete: ?() => mixed,
   ): rxjs$Subscription;
 
   static combineLatest<A, B>(
     a: rxjs$Observable<A>,
-    resultSelector: (a: A) => B,
+    resultSelector: ?(a: A) => B,
```
Comment (nmote) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js]: Could you add tests to make sure that this still works as expected, with and without the `resultSelector` argument?

**Comment by taichi** [comment]
Great work! i want to use this pull request.

**Comment by Kriegslustig** [comment]
@taichi Sorry, I'll probably get around to fixing these issues tonight.

File: definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js (hunk 1 above)
Comment (Kriegslustig) [inline definitions/npm/rxjs_v5.0.x/flow_v0.29.0-/rxjs_v5.0.x.js]: Removed it.

**Comment by marudor** [comment]
This hasn't been updated for quite a while and I can't add commits to the PR.
If this is still relevant please submit a new PR.

**Pull request closed without merging**